src/inmanta/server/services/resourceservice.py

```diff
@@ -416,6 +417,108 @@ async def mark_deployed(
             connection=connection,
         )
 
+    async def _update_deploy_state(
```

Comment:
This method now triplicates the functionality of `resource_action_update` and `deploy_start/deploy_done`

I do this because:
- `resource_action_update` is overloaded and should go
- `deploy_start/deploy_done` do half the work, this one is optimized to do both halves 

But:
- I now allow all types of  status updates, but test coverage is limited to failure. Should I just make this method refuse other updates (and strip it down further)?

Reply:
If I understand correctly (big "if"), `resource_action_update` is currently used for updates outside of the deploy path and we want to clean it up some day. Until that day, we don't expect any new use cases to arise for a state update like this one.

If this is correct, I have a minor preference to keep this method minimal for now, and to unify it with the greater whole once we address the `resource_action_update` bloat issue, because only then will we clearly see the full picture, and test the full picture.

Reply:
done